packages/plugin-js-packages/src/lib/runner/audit/constants.ts

```diff
@@ -47,16 +81,17 @@ const yarnv2EnvironmentOptions: Record<DependencyGroup, string> = {
   optional: '',
 };
 
+const pnpmDependencyOptions: Record<DependencyGroup, string[]> = {
+  prod: ['--prod', '--no-optional'],
```

Comment:
While the object can be useful for optional arguments or arguments with values that need pre-processing, here I am listing hardcoded arguments from the package manager documentation. Their values will not change so I do not see the value of adding an extra wrapper function.

Reply:
I personally prefer to see `--prod --no-optional`, because it's very transparent in how it corresponds to `pnpm` command line arguments, which I can look up in their official docs. Having `objToCliArgs({ prod: true, optional: false })` makes this relationship less clear, and I don't really see what advantage it brings :man_shrugging:

Reply:
As the other comments are resolved, I am going to merge the PR. That does not mean I am not open to a discussion about this but it is something that does not affect the functionality and can be changed later. Feel free to bring it up on Slack.